docs/deployment-examples/custom-scripts/scripts-in-packages/index.md

```diff
@@ -11,19 +11,37 @@ You can add any of the following script files in any of the scripting languages
 
-As mentioned above, you can create a file named `DeployFailed.<ext>`, which will be invoked if the package deployment fails. Our blog post about this feature [describes how DeployFailed.<ext> works](https://octopus.com/blog/deployfailed).
+## Including the Scripts in the Package
 
-As of 4.1.10, you can prevent the running of scripts in packages by adding the `Octopus.Action.Package.RunScripts` variable to your project (scoped as needed) and setting it to `false`.
+1. Create the scripts you want Octopus to execute during the step.
+2. Name each script to match the naming convention depending when you want the script to execute.
+3. Include these scripts at the root of your package. Octopus will not search subdirectories.
 
 :::hint
-**Script Support on Deployment Targets**
-Of course, Bash scripts will only be supported on Linux / OSX Targets and PowerShell and Script CS will only run on Windows. So ensure you've selected the correct language for your deployment target
+Avoid duplicate scripts into your package, like `PreDeploy.sh` and `PreDeploy.ps1`, hoping Octopus will choose the right script on your behalf. Octopus will try to execute both of these scripts during the pre-deploy phase of the step since both bash and PowerShell are cross-platform runtimes. This might lead to surprising behavior.
 :::
 
+## Running a Script When A Step Fails
 
-Make sure that the scripts are included in your package. If you are using OctoPack for an ASP.NET web application, you'll need to make sure the file is marked as **Build Action =** **Content**.
+You can create a file named `DeployFailed.<ext>`, which will be invoked if the step fails. Our blog post about this feature [describes how DeployFailed.<ext> works](https://octopus.com/blog/deployfailed).
+
+## Disabling this Convention
+
+You can prevent Octopus from automatically running scripts in packages by adding the `Octopus.Action.Package.RunScripts` variable to your project and setting it to `false`. You can scope the value of this variable to suit your needs.
+
+## Scripts in Package Steps {#scripts-in-package-steps}
+
+Rather than embed scripts in packages, you can also define scripts within the package step definition in Octopus. This is a feature that can be enabled on package steps by clicking **CONFIGURE FEATURES** and selecting **custom deploy scripts**.
+
+When enabled, you will see **Configuration Scripts** under the features section of the process definition.
+
+## Troubleshooting
+
+Make sure the scripts are located in the root of your package.
+
+Make sure the scripts are actually included in your package. Extract your package and inspect the contents to make sure the scripts are included as you expect. For example, if you are using OctoPack for an ASP.NET web application, you'll need to make sure the file is marked as **Build Action =** **Content**.
```

Comment:
I changed the formatting here. As I'm not sure the emphasis would work the way you expect it :)